lib/cn/kvset.c

```diff
@@ -1088,7 +1056,7 @@ _kvset_destroy(struct kvset *ks)
     cleanup_kblocks(ks);
 
     if (ks->ks_deleted && !atomic_read(&ks->ks_delete_error))
-        cndb_txn_ack_d(ks->ks_cndb, ks->ks_delete_txid, ks->ks_tag, ks->ks_cnid);
+        cndb_record_kvset_del_ack(ks->ks_cndb, ks->ks_delete_txn, ks->ks_delete_cookie);
```

Comment:
I really don't like using ks_deleted as a bool here since I see it's being set to some opaque values above (DEL_KEEPV or DEL_ALL).  So now I have to go figure that out to see on what conditions we call del_ack()...
I realized this predates you :-)

Reply:
Hmm, I can change that to `if (ks->ks_deleted != DEL_NONE)`

Reply:
If that's the correct logic then IMO it's infinitely more clear!

Reply:
I'm w/ Greg on this one.